docs/workflow-lifecycle.md

```diff
@@ -0,0 +1,239 @@
+   - An Activity task is added in the Matching service.
+
+```mermaid
+sequenceDiagram
+Worker ->> Frontend: RespondWorkflowTaskCompleted<br>[ScheduleActivityTask]
+Frontend->> History: RespondWorkflowTaskCompleted<br>[ScheduleActivityTask]
+History ->> Persistence: UpdateWorkflowExecution
+note over Persistence: append History Events: WorkflowTaskCompleted, ActivityTaskScheduled<br>update MutableState and add Transfer Task
+Persistence ->> History: Update Succeed
+History->>Frontend: Respond Succeed
+Frontend->>Worker: Respond Succeed
+loop QueueProcessor
+    History->>Persistence: GetHistoryTasks
+		History->>History: ProcessTask
+		History->>Matching: AddActivityTask
+end
+```
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- History service [`ScheduleActivityTask` command handler](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/workflow_task_handler.go#L338)
+
+</details>
+<br>
+
+---
+
+<br>
+
+**4. The Worker dequeues the Activity task and executes the activity**
+
+```mermaid
+sequenceDiagram
+Worker->>Frontend: PollActivityTask
+Frontend->>Matching: PollActivityTask
+History->>Matching: AddActivityTask (step 3 above)
+Matching->>History: RecordActivityStarted
+History->>Persistence: UpdateWorkflowExecution
+note over Persistence: append History Event: ActivityTaskStarted<br>update Mutable State, add Timer Task (activity task timeout)
+Persistence->>History: Update succeed
+History->>Matching: Record Succeed
+Matching->>Frontend: ActivityTask
+Frontend->>Worker: ActivityTask
+Worker->>Worker: Execute activity
+```
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- History service [`RecordActivityTaskStarted` handler](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/handler.go#L287)
+
+</details>
+<br>
+
+---
+
+<br>
+
+5. **The Worker sends `RespondActivityTaskCompleted`, containing the activity result**
+   - A Workflow Task is added in the Matching service
+
+```mermaid
+sequenceDiagram
+Worker->>Frontend: RespondActivityTaskCompleted
+Frontend->>History: RespondActivityTaskCompleted
+History->>Persistence: UpdateWorkflowExecution
+note over Persistence: append History Events:<br>ActivityTaskCompleted  (w/ activity result), WorkflowTaskScheduled<br>update Mutable State & add Transfer Task
+Persistence->>History: Update Succeed
+History->>Frontend: Respond Succeed
+Frontend->>Worker: Respond Succeed
+loop QueueProcessor
+    History->>Persistence: GetHistoryTasks
+		History->>History: ProcessTask
+		History->>Matching: AddWorkflowTask
+end
+```
+
+<details>
+<summary><i>Code entrypoints</i></summary>
+
+- History service [`RespondActivityTaskCompleted` handler](https://github.com/temporalio/temporal/blob/ef49189005b5323c532264287af6c08a447aab8a/service/history/handler.go#L361)
+
+</details>
+<br>
+
+---
+
+<br>
+
+6. **The Worker dequeues the Workflow Task**
```

Comment:
I know this is perhaps oversimplified (since we're not talking about replay), but:

```suggestion
6. **The Worker running the workflow dequeues the Workflow Task**
```

Reply:
Thanks, changed to "A worker dequeues"